src/software/jetson_nano/services/motor.cpp

```diff
@@ -177,11 +167,17 @@ void MotorService::setUpMotors()
     checkDriverFault(DRIBBLER_MOTOR_CHIP_SELECT);
     startController(DRIBBLER_MOTOR_CHIP_SELECT, true);
     tmc4671_setTargetVelocity(DRIBBLER_MOTOR_CHIP_SELECT, 0);
-}
 
+    checkEncoderConnections();
 
-bool MotorService::checkDriverFault(uint8_t motor)
+    is_initialized_ = true;
+}
+
+MotorService::MotorFaultIndicator MotorService::checkDriverFault(uint8_t motor)
 {
+    bool drive_enabled = true;
```

Comment:
nima: If drive is not enabled for some motor, do we want to reset it? Or is the plan to manually reset it by restarting thunderloop?

Reply:
yeah my code doesn't handle this other than just notifying thunderscope

Reply:
I think thunderloop should be responsible for resetting on faults, what do you think? @nimazareian

Reply:
@nimazareian

Reply:
I think we should reset the motor, but it might also be a good idea to discuss the implications of resetting a faulty motor driver instantly with Elec, and if there are any risks involved.

Reply:
arun: we'll handle this in a new PR